lib/ObjWriter/objwriter.cpp

```diff
@@ -631,6 +728,44 @@ extern "C" void EmitDebugFunctionInfo(ObjectWriter *OW,
   OST.EmitCVLinetableDirective(OW->FuncId++, Fn, FnEnd);
 }
 
+extern "C" void EmitDebugFunctionInfo(ObjectWriter *OW,
+                                      const char *FunctionName,
+                                      int FunctionSize) {
+  assert(OW && "ObjWriter is null");
+  auto *AsmPrinter = &OW->getAsmPrinter();
+  auto &OST = static_cast<MCObjectStreamer &>(*AsmPrinter->OutStreamer);
+  MCContext &OutContext = OST.getContext();
+  const MCObjectFileInfo *MOFI = OutContext.getObjectFileInfo();
+
+  if (MOFI->getObjectFileType() == MOFI->IsCOFF) {
+    EmitCVDebugFunctionInfo(OW, FunctionName, FunctionSize);
+  } else {
+    // TODO: Should convert this for non-Windows.
+  }
+}
+
+extern "C" void EmitDebugVar(ObjectWriter *OW, char *name, int typeIndex,
+                             bool isParm, int rangeCount, char *ranges) {
+  assert(OW && "ObjWriter is null");
+
+  if (rangeCount == 0)
+    return;
+
+  // Interpret ranges as an array of NativeVarInfo structs.
+  ICorDebugInfo::NativeVarInfo *varInfos =
+      (ICorDebugInfo::NativeVarInfo *)ranges;
```

Comment:
I think type cast is not needed. Instead, you could declare the type for ranges as "NativeVarInfo*" instead of "char *" in the function argument since you are marshalling the array of struct.